tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/PublicClientApplicationTests.cs

```diff
@@ -697,27 +697,63 @@ public async Task AcquireTokenSilentScopeAndEmptyCacheTestAsync()
+                    MsalTestConstants.ClientId,
+                    MsalTestConstants.AuthorityCommonTenant)
                 {
                     ValidateAuthority = false
                 };
 
-            var cache = new TokenCache()
-            {
-                ClientId = MsalTestConstants.ClientId
-            };
+                PublicClientApplication app2 = new PublicClientApplication(
+                    serviceBundle,
+                    MsalTestConstants.ClientId,
+                    MsalTestConstants.AuthorityTestTenant)
+                {
+                    ValidateAuthority = false
+                };
 
-            app.UserTokenCache = cache;
-            TokenCacheHelper.PopulateCache(cache.TokenCacheAccessor);
-            try
-            {
-                AuthenticationResult result = await app.AcquireTokenSilentAsync(MsalTestConstants.Scope.ToArray(),
-                        new Account(MsalTestConstants.UserIdentifier, MsalTestConstants.DisplayableId, null)).ConfigureAwait(false);
-            }
-            catch (MsalClientException exc)
-            {
-                Assert.AreEqual(MsalClientException.MultipleTokensMatchedError, exc.ErrorCode);
+                var cache = new TokenCache()
+                {
+                    ClientId = MsalTestConstants.ClientId
+                };
+
+                app2.UserTokenCache = cache;
+                TokenCacheHelper.PopulateCache(cache.TokenCacheAccessor);
+
+                app.UserTokenCache = cache;
+                TokenCacheHelper.PopulateCache(cache.TokenCacheAccessor);
+
+                httpManager.AddInstanceDiscoveryMockHandler();
+                httpManager.AddMockHandlerForTenantEndpointDiscovery(MsalTestConstants.AuthorityTestTenant);
+
+                httpManager.AddMockHandler(
+                    new MockHttpMessageHandler()
+                    {
+                        Method = HttpMethod.Post,
+                        ResponseMessage = MockHelpers.CreateSuccessTokenResponseMessage(
+                            MsalTestConstants.UniqueId,
+                            MsalTestConstants.DisplayableId,
+                            MsalTestConstants.Scope.ToArray())
+                    });
+
+                try
+                {
+                    AuthenticationResult result = await app.AcquireTokenSilentAsync(
+                                                      MsalTestConstants.Scope.ToArray(),
+                                                      new Account(
+                                                          MsalTestConstants.UserIdentifier,
+                                                          MsalTestConstants.DisplayableId,
+                                                          null)).ConfigureAwait(false);
+                }
+                catch (MsalClientException exc)
+                {
+                    Assert.AreEqual(MsalClientException.MultipleTokensMatchedError, exc.ErrorCode);
```

Comment:
Could you add an assert message explaining why you expect there 2 be this exception, i.e. why there are 2 tokens in the cache ?